src/bsoncxx/test/bson_builder.cpp

```diff
@@ -1702,36 +1702,65 @@ TEST_CASE("list builder with explicit type deduction", "[bsoncxx::builder::list]
         bson_append_int32(&array, "1", -1, 1);
         bson_append_array(&expected, "foo", -1, &array);
 
-        builder::list b{"foo", builder::array{"bar", 1}};
-        bson_eq_object(&expected, b.view().get_document().value);
+        builder::list::document b{"foo", builder::list::array{"bar", 1}};
+        bson_eq_object(&expected, b.extract().view());
 
         bson_destroy(&expected);
         bson_destroy(&array);
     }
 
-    SECTION("document") {
-        builder::list b;
-        auto kvp_regex =
-            Catch::Matches("(.*)must be list of key-value pairs(.*)", Catch::CaseSensitive::No);
-        REQUIRE_THROWS_WITH((b = builder::document{"foo", 1, 2}), kvp_regex);
-
-        auto type_regex =
-            Catch::Matches("(.*)must be string type(.*)int32(.*)", Catch::CaseSensitive::No);
-        REQUIRE_THROWS_WITH((b = builder::document{"foo", 1, 2, 4}), type_regex);
-    }
+    // compile-time error
```

Comment:
I left this here so later devs could confirm this snippet still throws a compile-time error upon uncommenting. This is similar to what's done on line 108.

Reply:
Update the test text so this comment isn't needed anymore. Also removed the surrounding test. Since we don't have implicit type deduction between arrays and documents, we don't need to test explicit type deduction. That's done in every other test.